src/components/validation/resources/event.rs

```diff
@@ -44,3 +45,93 @@ impl EventData {
         }
     }
 }
+
+/// An event used in a test case.
+/// It is important to have created the event with all fields, immediately after deserializing from the
+/// test case definition yaml file. This ensures that the event data we are using in the expected/actual
+/// metrics collection is based on the same event. Namely, one issue that can arise from creating the event
+/// from the event data twice (once for the expected and once for actual), it can result in a timestamp in
+/// the event which may or may not have the same millisecond precision as it's counterpart.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(try_from = "RawTestEvent")]
```

Comment:
Doesn't seem like this needs to actually be `try_from` and could be `from` instead.